go-controller/pkg/util/pod_annotation.go

```diff
@@ -261,24 +261,43 @@ func GetPodCIDRsWithFullMask(pod *v1.Pod) ([]*net.IPNet, error) {
 // GetAllPodIPs returns the pod's IP addresses, first from the OVN annotation
 // and then falling back to the Pod Status IPs. This function is intended to
 // also return IPs for HostNetwork and other non-OVN-IPAM-ed pods.
-func GetAllPodIPs(pod *v1.Pod) ([]net.IP, error) {
-	annotation, err := UnmarshalPodAnnotation(pod.Annotations, types.DefaultNetworkName)
-	if err == nil && annotation != nil {
+func GetAllPodIPs(pod *v1.Pod, nInfo NetInfo) ([]net.IP, error) {
```

Comment:
now we have util.DefaultNetworkNetNameInfo so it should be easier for the caller that only cares about default network?

I am reluctant to do what you suggested because it seems to me that GetPodIPsOfDefaultNetwork() and GetPodIPsOfNetwork() could share some code, and some functions shared by both default/secondary controllers can call the same function